src/main/kotlin/org/rust/ide/annotator/fixes/AddStructFieldsFix.kt

```diff
@@ -48,15 +49,7 @@ class AddStructFieldsFix(
         val body = structLiteral.structLiteralBody
         val fieldsToAdd = calculateMissingFields(body, decl)
         val defaultValueBuilder = RsDefaultValueBuilder(decl.knownItems, body.containingMod, RsPsiFactory(project), recursive)
-        val firstAdded = defaultValueBuilder.fillStruct(
-            body,
-            decl.fields,
-            fieldsToAdd
-        )
-
-        if (editor != null && firstAdded != null) {
-            val expr = firstAdded.expr
-            if (expr != null) editor.caretModel.moveToOffset(expr.textOffset)
-        }
+        val addedFields = defaultValueBuilder.fillStruct(body, decl.fields, fieldsToAdd)
+        editor?.buildAndRunTemplate(body, addedFields.mapNotNull { it.expr })
```

Comment:
Add `CodeStyleManager.getInstance(project).reformat(structLiteral)` before `buildAndRunTemplate` call to avoid `com.intellij.psi.PsiInvalidElementAccessException`